### ISSUE TITLE
Disabling godot script editor as an engine feature doesn't disable it from opening files automatically.

### ISSUE THREAD

**Issue by PowerfulBacon** [godotengine/godot#91659]
### Tested versions

- Tested and reproducible in v4.2.1.stable.mono.official

### System information

Godot v4.2.1.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 3GB (NVIDIA; 31.0.15.3623) - Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz (4 Threads)

### Issue description

When disabling the godot engine's script editor through Editor > Manage Editor Features, the editor continues to open up files automatically. This is particularly annoying if you are working with an external IDE, as you will continue to get popups from script files asking if they should reload.

When switching between scenes, the script editor will randomly open code files. This is annoying in of itself, but when the script editor is disabled this makes it impossible to go and close those scripts.

### Steps to reproduce

- Create a script file
- Create 2 scenes
- Add the new script to the root of the first scene.
- Close the script in the script editor if it is open.
- Switch between different scene tabs and the file will re-open itself.
- Close the file in the script editor
- Go to editor > manage editor features
- Disable the script editor
- Switch between scenes
- Edit the script in an external editor
- A popup will appear asking what action you would like to take.

### Minimal reproduction project (MRP)

[Reproduction.zip](https://github.com/godotengine/godot/files/15234955/Reproduction.zip)

**Comment by AThousandShips** [comment]
Do you have distraction free mode active? If so this would be a duplicate of:
* https://github.com/godotengine/godot/issues/90654

And should be fixed in 4.3 by:
* https://github.com/godotengine/godot/pull/91168

**Comment by PowerfulBacon** [comment]
> Do you have distraction free mode active?

No, this happens with distraction free mode is enabled or disabled. This happens because when you switch to another scene, if the root node has a script attached to it then it will automatically open in the script editor. This is really annoying if you are using an external IDE for editing scripts as it will constantly ask you if you want to reload or overwrite the changes, and causes loss of data ocassionally even if you reload.

**Comment by kitbdev** [comment]
Can reproduce in Godot v4.2.1.stable and master v4.3.dev.custom_build [2042420bd] - Windows 10.0.22631
I don't think #90654 is related.

It looks like the Script Editor opens it and keeps working even if it is disabled.

What external editor are you using? 
I can only get it to happen with certain ones like notepad but not VSCode.

Try turning off this Editor Setting that opens the root node's script: `text_editor/behavior/files/open_dominant_script_on_scene_change`

Also, there is an external editor setting that doesn't have this issue: `text_editor/external/use_external_editor` 
But this requires the Script Editor EditorFeature to be enabled to work.

**Comment by PowerfulBacon** [comment]
With `open_dominant_script_on_scene_change` disabled, the issue no longer happens as scripts are not opened in the background anymore.

I am using visual studio 2022, however I believe this will work with all editors since the message pops up whenever the file is edited while it is open in the Godot script editor. The issue is that even if the script editor is disabled, it will continue to open up the dominant script on scene change and while files are open in Godot's script editor, the file change message will continue to appear when you modify those files.
